src/main/java/hudson/plugins/scm/koji/model/RPM.java

```diff
@@ -86,6 +92,29 @@ public boolean hasUrl() {
         return url != null;
     }
 
+    public void setHashSum(File file) throws IOException, NoSuchAlgorithmException {
```

Comment:
please don't use setter that do so performance heavy operations. It's very nasty side effect. Setter should by conventions just set the value. You can calculate the hash and then set it to RPM.
Also, RPM is domain entity and this logic definitely should not be in this class.